### PR TITLE
Addressing leftovers after the merge of "SPI on KCP/CPS"

### PR DESCRIPTION
**Pull request by skabashnyuk**
## Goal
Addressing leftovers after the merge of https://github.com/redhat-appstudio/infra-deployments/pull/715

### CODE REVIEW

File: hack/preview.sh [hunk 1]
```diff
@@ -39,12 +39,30 @@ if [ -n "$MY_GITHUB_ORG" ]; then
     $ROOT/hack/util-set-github-org $MY_GITHUB_ORG
 fi
 
+
+echo
+echo -n "Waiting for 'spi-system' namespace to exist: "
+while ! kubectl get namespace spi-system --kubeconfig ${KCP_KUBECONFIG} &> /dev/null ; do
```
Comment (Michkov) [inline hack/preview.sh:45]: argoCD is not applied yet, so the spi-system namespace will not be created. The current code is creating the namespace and route, which is then updated by argoCD

File: argo-cd-apps/base/spi.yaml [hunk 2]
```diff
@@ -41,7 +41,7 @@ spec:
             duration: 15s # the amount to back off. Default unit is seconds, but could also be a duration (e.g. "2m", "1h")
 ---
 apiVersion: argoproj.io/v1alpha1
-kind: ApplicationSet
+kind: Application
 metadata:
   name: spi-vault
 spec:
```
Comment (Michkov) [inline argo-cd-apps/base/spi.yaml:47]: generators and template should be removed

Comment (Michkov) [reply argo-cd-apps/base/spi.yaml:47]: should be similar to https://github.com/redhat-appstudio/infra-deployments/blob/pre-kcp/argo-cd-apps/base/spi.yaml

also https://github.com/redhat-appstudio/infra-deployments/blob/main/argo-cd-apps/overlays/development/repo-overlay.yaml#L34-L50 should be updated

File: argo-cd-apps/base/spi.yaml [hunk 3]
```diff
@@ -41,42 +41,34 @@ spec:
...
-      name: spi-vault
-    spec:
-      project: default
+  project: default
 
-      source:
-        path: components/spi/vault
-        repoURL: https://github.com/redhat-appstudio/infra-deployments.git
-        targetRevision: main
+    source:
```
Comment (Michkov) [inline argo-cd-apps/base/spi.yaml:50]: `source`(and other elements on the level) must be same level as `project`

**Review by Michkov — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**